### PR TITLE
Remove --autogen-autoloader-pbal-non-annotated flag and enable PBAL for all non-legacy packages by default

### PR DESCRIPTION
**Pull request by aadi-stripe**
Removes the `--autogen-autoloader-pbal-non-annotated` flag as it isn't required anymore, and also removes the `autoloader_compatibility 'strict'` annotation. Path-based autoloading will now be enabled for all non-legacy packages by default.

### Motivation

Migration of Stripe codebase to Zeitwerk autoloader.

### Test plan

See included automated tests. Tested on Stripe codebase.

### CODE REVIEW

File: test/cli/package-special-allowed-dsls/foo/__package.rb [hunk 1]
```diff
@@ -3,6 +3,4 @@
 class Project::Foo < PackageSpec
   restrict_to_service Project::Foo
   restrict_to_service Project::Bar
-
-  autoloader_compatibility 'strict'
```
Comment (jez) [inline test/cli/package-special-allowed-dsls/foo/__package.rb:7]: Do you think it's worth keeping at least one mention of `autoloader_compatibility 'strict'` in the test suite to assert that we produce an error on it?

File: packager/packager.cc [hunk 2]
```diff
@@ -1029,22 +1021,17 @@ struct PackageInfoFinder {
             auto compatibilityAnnotation = compatibilityAnnotationLit->asString();
             if (compatibilityAnnotation != core::Names::strict() && compatibilityAnnotation != core::Names::legacy()) {
                 if (auto e = ctx.beginError(send.loc, core::errors::Packager::InvalidConfiguration)) {
-                    e.setHeader("Argument to `{}` must be either 'strict' or 'legacy'", send.fun.show(ctx));
+                    e.setHeader("Argument to `{}` can only be 'legacy'", send.fun.show(ctx));
```
Comment (jez) [inline packager/packager.cc:1024]: Up to you, but you could consider specializing this error if `fun` is `strict` with a custom message and/or autocorrect.

**Review by jez — APPROVED** [review]

**Comment by aadi-stripe** [comment]
@jez https://github.com/sorbet/sorbet/pull/6887/commits/abbd69684204b719ba2b38a2720c430a2a6da9a0

File: packager/packager.cc [hunk 3]
```diff
@@ -1027,24 +1019,26 @@ struct PackageInfoFinder {
...
 
             auto compatibilityAnnotation = compatibilityAnnotationLit->asString();
-            if (compatibilityAnnotation != core::Names::strict() && compatibilityAnnotation != core::Names::legacy()) {
+            if (compatibilityAnnotation != core::Names::legacy()) {
                 if (auto e = ctx.beginError(send.loc, core::errors::Packager::InvalidConfiguration)) {
-                    e.setHeader("Argument to `{}` must be either 'strict' or 'legacy'", send.fun.show(ctx));
+                    if (compatibilityAnnotation == core::Names::strict()) {
+                        e.setHeader("strict has been deprecated as an argument to `{}`. If you wish to mark your "
+                                    "package as strictly path-based-autoloading compatible, do not provide an "
+                                    "autoloader_compatibility annotation",
```
Comment (jez) [inline packager/packager.cc:1027]: Don't put multiple sentences in a header

If you want to add more context, provide all the subsequent context with an `addErrorNote`.

Comment (aadi-stripe) [reply packager/packager.cc:1027]: Ah sorry, sure.

**Comment by aadi-stripe** [comment]
@jez fixed

**Comment by aadi-stripe** [comment]
> @jez fixed. I used `addErrorLine`. Although perhaps I should have used `addErrorNote` as you said...lemme try that.

Oh yeah, that's better.

**Comment by aadi-stripe** [comment]
> > @jez fixed. I used `addErrorLine`. Although perhaps I should have used `addErrorNote` as you said...lemme try that.
> 
> Oh yeah, that's better.

Fixed

**Review by jez — APPROVED** [review]

**Pull request merged**